### DISCUSSION TITLE
[Feature] Ikemen Go with its own customizable select stage screen

### DISCUSSION THREAD

**Discussion by Paul45mec** [ikemen-engine/Ikemen-GO#940] [Feature Request]
### Request for module or a new implement for a customizable stage select screen.

Below is the model with some descriptions that could help in the creation of a new module to expand the edition of this type of feature. 

Being customizable so that we can put the animations we want in the edition as we do with the other screens.

> **[Stage Select Screen]**

;Transition to separate the Character Select screen to the Stage Select screen.

obs: all the attributes like offset,layerno and scale are considered.

;Portrait for the stage (sprite or animation)

**stage.portrait.spr =
stage.portrait.anim =**

;Portrait for the stage background (sprite or animation)

**stage.portrait.bg.spr =
stage.portrait.bg.anim =**

;Arrow icon to indicate right/left (sprite or animation)

**stage.arrow.spr =
stage.arrow.anim =**

;Stage voice (Narrator of the stage name, sound added in system.snd)

**stage.snd = 3000.0**

;music (Own soundtrack for this screen)

**stage.bgm = sound/stage.mp3**

; stage SFX

**stage.move.snd = 100.0
stage.done.snd = 100.1
cancel.snd = 100.2**

; used to put the name of the stage

**stage.name.text = "%s"**
**stage.name.font=**

; used to put a story for the stage

**stage.description.text= "%s"**
**stage.description.font=**

**Comment by MUGEN-Slayer** [comment]
Currently, the stage select parameters practically already work as a separate screen. The only problem would be if any element of the select screen used "layerno = 1" because it would appear overlapping on the stage selection. But from what I read a while ago, one of the devs is working on how things work on the select screen so we have control over the priority of the sprites.
When that is achieved (taking issue #721 into account), I suppose it would be easier to keep things as they are, but with extra additions like:
 - "Fadein.anim" and "fadeout.anim" when stage selection is triggered.
 - More bgs for "stage.portrait" (being "stage.portrait.bg." = "stage.portrait.bg0."):
stage.portrait.bg1.
stage.portrait.bg2.
... And so on. 
With the additional parameters of "tile", "tilespacing", "velocity" and "window",
- "Stage.snd" and "stage.bgm" sounds good. Then as soon as we entered stage selection mode, if "stage.bgm" is assigned, the select screen music stops.
- "Stage.description" isn't bad either.

**Comment by Paul45mec** [comment]
> Currently, the stage select parameters practically already work as a separate screen. The only problem would be if any element of the select screen used "layerno = 1" because it would appear overlapping on the stage selection. But from what I read a while ago, one of the devs is working on how things work on the select screen so we have control over the priority of the sprites. When that is achieved (taking issue #721 into account), I suppose it would be easier to keep things as they are, but with extra additions like:
> 
> * "Fadein.anim" and "fadeout.anim" when stage selection is triggered.
> * More bgs for "stage.portrait" (being "stage.portrait.bg." = "stage.portrait.bg0."):
>   stage.portrait.bg1.
>   stage.portrait.bg2.
>   ... And so on.
>   With the additional parameters of "tile", "tilespacing", "velocity" and "window",
> * "Stage.snd" and "stage.bgm" sounds good. Then as soon as we entered stage selection mode, if "stage.bgm" is assigned, the select screen music stops.
> * "Stage.description" isn't bad either.

You're completely right! 
The layerno = 1 is a problem and how the stage sprite or anim forcibly appears over all, the transition doesn't get to appear over all the elements first and only after which stages should appear.  

I think the next updatings will help us to solve many questions.
